### DISCUSSION TITLE
What is the purpose of IPipelineBehavior in the following code?

### DISCUSSION THREAD

**Discussion by mahedee** [OpenCodeFoundation/eSchool#349] [Q&A] [answered]
I don't understand the meaning of following code. Why do you use the following code in Startup.cs of Enrollment.API project?
```csharp
  public void ConfigureServices(IServiceCollection services)
  {
            services.AddTransient(typeof(IPipelineBehavior<,>), typeof(LoggingBehavior<,>));
  }
```

**Comment by ratanparai** [comment, marked as answer]
Here is a good article to read about the use cases of `IPipelineBehavior` - https://lurumad.github.io/cross-cutting-concerns-in-asp-net-core-with-meaditr

The code snippet inject the LoggingBehavior to the service so that `MediatR` can find our logging behavior before calling the handlers

I hope this will help.

**Reply by mahedee** [reply]
Thank you. I understand now.
